node-graph/gcore/src/graphic_element/renderer.rs

```diff
@@ -271,7 +272,12 @@ pub fn to_transform(transform: DAffine2) -> usvg::Transform {
 pub trait GraphicElementRendered {
 	fn render_svg(&self, render: &mut SvgRender, render_params: &RenderParams);
 	fn bounding_box(&self, transform: DAffine2) -> Option<[DVec2; 2]>;
-	fn add_click_targets(&self, click_targets: &mut Vec<ClickTarget>);
+	fn add_click_targets(&self, _click_targets: &mut HashMap<NodeId, Vec<ClickTarget>>, _element_id: Option<NodeId>) {}
+	fn add_upstream_click_targets(&self, _click_targets: &mut Vec<ClickTarget>) {}
+	// TODO: Store all click targets in a vec which contains the AABB, click target, and path
+	// fn add_click_targets(&self, click_targets: &mut Vec<([DVec2; 2], ClickTarget, Vec<NodeId>)>, current_path: Option<NodeId>) {}
+	fn add_vector_modify(&self, _vector_modify: &mut HashMap<NodeId, VectorData>, _element_id: Option<NodeId>) {}
+	fn add_footprints(&self, _footprints: &mut HashMap<NodeId, (Footprint, DAffine2)>, _footprint: Footprint, _element_id: Option<NodeId>) {}
```

Comment:
We should think about if it makes sense to collect all of these in one go. Doing multiple identiacal tree traversals would be wasted compute

Reply:
This could also reduce the amount of structural code duplication making the code easier to maintain